frontend/src/metabase/query_builder/components/QuestionActivityTimeline.jsx

```diff
@@ -101,39 +77,43 @@ export function QuestionActivityTimeline({
     const revisionEvents = getRevisionEventsForTimeline(revisions, {
       currentUser,
       canWrite,
+    }).map(revisionEvent => {
+      if (revisionEvent.isRevertable) {
+        const { title, revision } = revisionEvent;
+        const newTitle = (
+          <>
+            {title}
+            <Tooltip tooltip={t`Revert`} placement="bottom">
+              <RevertButton
+                icon="revert"
+                onlyIcon
+                borderless
+                onClick={() => revertToRevision(revision)}
+                data-testid="question-revert-button"
+              />
+            </Tooltip>
+          </>
+        );
+        return { ...revisionEvent, title: newTitle };
```

Comment:
I started making this change with the mindset of "Make adjustments to the existing structure", but I do wonder if it makes more sense to update `getRevisionEventsForTimeline` instead.

Reply:
+1 to Kelvin